nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/ConsumeKafka.java

```diff
@@ -0,0 +1,213 @@
+     */
+    @Override
+    protected void rendezvousWithKafka(ProcessContext context, ProcessSession processSession) throws ProcessException {
+        ConsumerRecords<byte[], byte[]> consumedRecords = this.kafkaResource.poll(100);
+        if (consumedRecords != null && !consumedRecords.isEmpty()) {
+            long start = System.nanoTime();
+            FlowFile flowFile = processSession.create();
+            final AtomicInteger messageCounter = new AtomicInteger();
+
+            for (final ConsumerRecord<byte[], byte[]> consumedRecord : consumedRecords) {
+                flowFile = processSession.append(flowFile, new OutputStreamCallback() {
+                    @Override
+                    public void process(final OutputStream out) throws IOException {
+                        if (messageCounter.getAndIncrement() > 0) {
+                            out.write(ConsumeKafka.this.demarcatorBytes);
+                        }
+                        out.write(consumedRecord.value());
+                    }
+                });
+            }
+
+            this.releaseFlowFile(flowFile, processSession, flowFile.getAttributes(), start, messageCounter.get() - 1);
+            this.kafkaResource.commitSync();
+        } else {
+            context.yield();
+        }
+    }
+
+    /**
+     * Builds and instance of {@link KafkaConsumer} and subscribes to a provided
+     * topic.
+     */
+    @Override
+    protected KafkaConsumer<byte[], byte[]> buildKafkaResource(ProcessContext context, ProcessSession session) throws ProcessException {
+        this.demarcatorBytes = context.getProperty(MESSAGE_DEMARCATOR).evaluateAttributeExpressions().getValue().getBytes(StandardCharsets.UTF_8);
+        this.topic = context.getProperty(TOPIC).evaluateAttributeExpressions().getValue();
+
+        Properties kafkaProperties = this.buildKafkaProperties(context);
+        kafkaProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+        kafkaProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+
+        KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(kafkaProperties);
+        consumer.subscribe(Collections.singletonList(this.topic));
+        return consumer;
+    }
+
+    /**
+     * Will release flow file. Releasing of the flow file in the context of this
+     * operation implies the following:
+     *
+     * If Empty then remove from session and return If has something then
+     * transfer to {@link #REL_SUCCESS}
+     */
+    private void releaseFlowFile(FlowFile flowFile, ProcessSession session, Map<String, String> attributes, long start, int msgCount) {
+        flowFile = session.putAllAttributes(flowFile, attributes);
+        long millis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
+        session.getProvenanceReporter().receive(flowFile, "kafka://" + this.topic, "Received " + msgCount + " Kafka messages", millis);
```

Comment:
I think IP bootstrap servers IPs is/are missing if the URI: kafka://<IPs/Ports>/topic, no?